modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/loader/XmlExtensionLoaderDelegate.java

```diff
@@ -272,7 +281,54 @@ private Document getModuleDocument(ExtensionLoadingContext context, URL resource
+  private ExtensionModel createTnsExtensionModel(URL resource, Set<ExtensionModel> extensions) throws IOException {
+    return new ExtensionModelFactory().create(
+                                              new DefaultExtensionLoadingContext(declareTns(resource, extensions),
+                                                                                 currentThread().getContextClassLoader(),
+                                                                                 new NullDslResolvingContext()));
+  }
+
+  private ExtensionDeclarer declareTns(URL resource, Set<ExtensionModel> extensions) throws IOException {
+    final ByteArrayOutputStream resultStream = new ByteArrayOutputStream();
+    try {
+      final Source xslt = new StreamSource(getClass().getClassLoader().getResourceAsStream(TRANSFORMATION_BODY_REMOVAL));
+      final Source moduleToTransform = new StreamSource(resource.openStream());
+      TransformerFactory.newInstance()
+          .newTransformer(xslt)
+          .transform(moduleToTransform, new StreamResult(resultStream));
+    } catch (TransformerException e) {
+      throw new MuleRuntimeException(
+                                     createStaticMessage(format("There was an issue transforming the stream for the resource %s while trying to remove the content of the <body> element to generate an XSD",
+                                                                resource.getFile())),
+                                     e);
+    }
+
+    final Document transformedModuleDocument = schemaValidatingDocumentLoader(NoOpXmlErrorHandler::new)
+        .loadDocument(extensions, resource.getFile(), new ByteArrayInputStream(resultStream.toByteArray()));
+    final ExtensionDeclarer extensionDeclarer = new ExtensionDeclarer();
+    loadModuleExtension(extensionDeclarer, resource, transformedModuleDocument);
+    return extensionDeclarer;
+  }
+
+  private ComponentModel getModuleComponentModel(URL resource, Document moduleDocument) {
+    XmlApplicationParser xmlApplicationParser =
+        new XmlApplicationParser(new SpiServiceRegistry(), singletonList(currentThread().getContextClassLoader()));
+    Optional<ConfigLine> parseModule = xmlApplicationParser.parse(moduleDocument.getDocumentElement());
+    if (!parseModule.isPresent()) {
+      // This happens in org.mule.runtime.config.spring.dsl.processor.xml.XmlApplicationParser.configLineFromElement()
+      throw new IllegalArgumentException(format("There was an issue trying to read the stream of '%s'", resource.getFile()));
+    }
+    ComponentModelReader componentModelReader =
+        new ComponentModelReader(new DefaultConfigurationPropertiesResolver(empty(),
+                                                                            new SystemPropertiesConfigurationProvider()));
+    // TODO MULE-12291: we would, ideally, leave either the relative path (modulePath) or the URL to the current config file,
```

Comment:
This was already fix for application configuration files, can you fix it for modules?